src/services/hsm.service.js

```diff
@@ -6,16 +6,210 @@
  */
 
 const { hsm } = require('@dinamonetworks/hsm-dinamo');
-const _ = require('lodash');
+const hathorLib = require('@hathor/wallet-lib');
 
 const { getConfig } = require('../settings');
 const { lock, lockTypes } = require('../lock');
 const { hsmBusyErrorMessage } = require('../helpers/constants');
 const { HsmError } = require('../errors');
 
+/* istanbul ignore next */
+/**
+ * After an error, the HSM lib has a runtime issue that will fail the next call
+ * To avoid this we can wait 500ms before continueing.
+ * This may be fixed in the near future.
+ * This method will return a promise that resolves after the given ms
+ * @param {number} ms
+ * @returns {Promise<void>}
+ */
+async function delay(ms) {
+  // eslint-disable-next-line no-promise-executor-return
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+/**
+ * Get the bip32 keyname
+ * @param {string} keyname
+ * @param {number} pathIndex
+ * @param {Object} [options={}]
+ * @param {number} options.index Derivation index for address level
+ */
+function getBip32Keyname(keyname, pathIndex, options = {}) {
+  const levels = [
+    () => keyname, // Root level
+    name => `${name}_bip44`,
+    name => `${name}_coinHTR`,
+    name => `${name}_htrAcct`,
+    name => `${name}_htrChange0`,
+    /**
+     * This unconventional naming is due to the 32 chars keyname restriction,
+     * we reserve 17 chars for context of which "*_HAddr_*" uses 7, leaving 10
+     * for the address index, this allows us to use all 2 billion indexes
+     * without resorting to hex conversion since 2 billion in base 10 has
+     * length 10.
+     */
+    (name, opts) => `${name}_HAddr_${opts.index}`,
+  ];
```

Comment:
I didn't understand the need for this array. Why don't you have only this?

```
const levels = [
  keyname,
  `${keyname}_bip44`,
  `${keyname}_coinHTR`,
  `${keyname}_htrAcct`,
  `${keyname}_htrChange0`,
  `${keyname}_HAddr_${options.index}`
];
```

Reply:
This new levels array will be created when the method is called, all levels will be formatted.
The issue is the last one, `${keyname}_HAddr_${options.index}` will be an error on all derivations from level 0 to 4 since we are acessing `index` from the undefined `options`.

Making them functions will only format the one we are requesting.

Reply:
I think the method is really confusing as it is now. If the problem is the undefined, we can use this. What do you think?

```
const levels = [
  keyname,
  `${keyname}_bip44`,
  `${keyname}_coinHTR`,
  `${keyname}_htrAcct`,
  `${keyname}_htrChange0`,
];

if (options.index) {
  levels.push(`${keyname}_HAddr_${options.index}`);
}
```

Reply:
Done